packages/contracts/contracts/BaseVerifier.sol

```diff
@@ -0,0 +1,80 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import {IVerifier} from './interfaces/IVerifier.sol';
+
+contract BaseVerifier {
+    IVerifier verifier;
+
+    uint256 public constant SNARK_SCALAR_FIELD =
+        21888242871839275222246405745257275088548364400416034343698204186575808495617;
+
+    struct EpochKeySignals {
+        uint256 revealNonce;
+        uint256 stateTreeRoot;
```

Comment:
These don't need to all be `uint256`. Try to use the smallest types possible so the EVM will use less memory to represent the struct.

e.d. `revealNonce` can be a `bool`, `nonce` can be a `uint8`, `attesterId` can be `uint160`, `epoch` `uint48`.

Reply:
Done, `EpochKeySignals` and `ReputationSignals` now uses the smallest types